### PR TITLE
Release Kedro 0.18.5

### LINKED ISSUE #2239: Release Kedro `0.18.5`
### Description

Release Kedro `0.18.5` which will contain lots of new features for configuration. The release depends on the following tickets to be finished:

- [x] BLOCKER: https://github.com/kedro-org/kedro/issues/2255
- [x] #1909 (Docs)
- [x] #2148 
- [x] #2170
- [x] #2225 

Initially we wanted to include the below issues as well, but the implementation turned out to be trickier than expected, so we'll take more time to investigate a solution and won't let it block the release.
- [ ] #2146 
- [ ] #2212

### PR DESCRIPTION
**Pull request by ankatiyar**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Development notes

Release Kedro 0.18.5 🎉  Resolves #2239 
## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -37,13 +37,15 @@ authors:
   given-names: Huong
 - family-names: Okwa
   given-names: Nero
+- family-names: Rodríguez
```
Comment (merelcht) [inline CITATION.cff:40]: While you're at it, you might also remove Cvetanka and Lim.

Comment (ankatiyar) [reply CITATION.cff:36]: Done!

File: CITATION.cff [hunk 2]
```diff
@@ -31,19 +29,19 @@ authors:
...
 - family-names: Milne
   given-names: Antony
-- family-names: Nechevska
-  given-names: Cvetanka
 - family-names: Nguyen
   given-names: Huong
 - family-names: Okwa
   given-names: Nero
+- family-names: Rodríguez
+  given-names: Juan Luis Cano
```
[the remark this answers is not in the document]
Comment (ankatiyar) [reply CITATION.cff]: Done, thanks for the correction 😄

**Review by merelcht — APPROVED** [review]
image

or maybe we do.. 👀 😏

**Pull request merged**